src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -481,22 +493,99 @@ private string GetCloudApplicationName(string? applicationName, UserDeploymentSe
                 throw new InvalidCliArgumentException(DeployToolErrorCode.InvalidCliArguments, "Found invalid CLI arguments");
             }
 
+            // validate and return the applicationName from the deployment settings if present.
             if (!string.IsNullOrEmpty(userDeploymentSettings?.ApplicationName))
             {
                 if (_cloudApplicationNameGenerator.IsValidName(userDeploymentSettings.ApplicationName))
                     return userDeploymentSettings.ApplicationName;
 
                 PrintInvalidApplicationNameMessage();
-                throw new InvalidUserDeploymentSettingsException(DeployToolErrorCode.UserDeploymentInvalidStackName, "Please provide a valid stack name and try again.");
+                throw new InvalidUserDeploymentSettingsException(DeployToolErrorCode.UserDeploymentInvalidStackName, "Please provide a valid cloud application name and try again.");
             }
 
+            return string.Empty;
+        }
+
+        // This method prompts the user to select a CloudApplication name for existing deployments or create a new one.
+        // If a user chooses to create a new CloudApplication, then this method returns string.Empty
+        private string AskForCloudApplicationNameFromDeployedApplications(List<CloudApplication> deployedApplications)
+        {
+            if (!deployedApplications.Any())
+                return string.Empty;
+
+            var title = "Select an existing AWS deployment target to deploy your application to.";
+
+            var userResponse =
+                        _consoleUtilities.AskUserToChooseOrCreateNew(
+                            deployedApplications.Select(x => x.DisplayName),
+                            title,
+                            askNewName: false,
+                            defaultChoosePrompt: Constants.CLI.PROMPT_CHOOSE_DEPLOYMENT_TARGET,
+                            defaultCreateNewLabel: Constants.CLI.CREATE_NEW_APPLICATION_LABEL);
+
+            if (userResponse.CreateNew)
+                return string.Empty;
+
+            // Since the selected option will be the display name, we need to extract the actual name out of it.
+            // Ex - DisplayName = "MyAppStack (CloudFormationStack)", ActualName = "MyAppStack"
+            var cloudApplicationName = userResponse.SelectedOption?.Split().FirstOrDefault() ?? string.Empty;

```

Comment:
I don't like that we are doing string manipulation here. We should use the CloudApplication Name.

Reply:
The AskUserToChooseOrCreateNew method does not take a separate key and value. Am I missing something? Also this piece of logic was already there. Its not something new that I am introducing as part of this PR.

Reply:
The is a `AskUserToChooseOrCreateNew` method that take in a `UserInputConfiguration` where you can specify a DisplaySelector to do what you are trying to achieve. I agree that you are not introducing anything new, but since you are modifying this code. If we can do better, we should.

Reply:
Changes this to use the `UserInputConfiguration`